core/src/cluster_info_vote_listener.rs

```diff
@@ -482,8 +482,10 @@ impl ClusterInfoVoteListener {
             );
             match confirmed_slots {
                 Ok(confirmed_slots) => {
-                    confirmation_verifier
-                        .add_new_optimistic_confirmed_slots(confirmed_slots.clone());
+                    confirmation_verifier.add_new_optimistic_confirmed_slots(
```

Comment:
Can I know more about the workload of the optimistic slots?  Specifically:

* Is the key (i.e., slot) roughly or strickly monotonically increasing over time?
* What is the ideal TTL policy of this column family?  I know it's currently implemented in the same way as other slot-based column families in this PR but just want to double-check.
* Looks like ledger tool is the only place where we read from the blockstore.  Do we plan to read from other places?

Reply:
> Can I know more about the workload of the optimistic slots? Specifically:
> 
> * Is the key (i.e., slot) roughly or strickly monotonically increasing over time?
yes, slot should only increase

> * What is the ideal TTL policy of this column family?  I know it's currently implemented in the same way as other slot-based column families in this PR but just want to double-check.
This data is used when coordinating a cluster restart to find the latest optimistic slot that validators can agree to restart from. Currently `optimistic_slot` ends up in the logs. If validators have disabled logging or accidentally deleted log files they would not have this data available. This is intended as a less error prone way of finding the latest `optimistic_slot` for each validator. A huge amount of history should not be necessary.

> * Looks like ledger tool is the only place where we read from the blockstore.  Do we plan to read from other places?
No, this is really intended to be used for recovery (most likely when the validator isn't running).

Reply:
Got it.  Thanks for the explanation.  So the change will make the recovery runs more smoothly (or in the near future, automatically pick up the correct confirmed slot from the db during the recovery.)

Reply:
In the event a cluster restart is necessary this should streamline the process for a validator to find their latest `optimistic_slot`. Currently, validators have to find their latest log files, grep through them for `optimistic_slot` and find the highest value. If we do want to implement restart automation then we'll have this data in blockstore.

Reply:
Let me think more about it.  Do we only want to keep the latest or a list of the most recent ones?

If we really just need one, maybe we can create a column family called "ledger_metadata" or whatever is suitable, and have the key equal to "latest_optimistic_slot", and the value is the slot id + the existing fields?

Reply:
We should retain more than just the latest as a common `optimistic_slot` may need to be found across validators.

I made the default in ledger-tool return only the latest to minimize confusion when validators are trying to compare values during a restart.